### PR TITLE
Update P2P Service to Handle Local Metadata

### LINKED ISSUE #5290: Modify Our Peerstore to accept ENRs as the Primary Node Identity
Currently each peer has their peer id as the primary basis for their node identity in our peer store. With the testnet restart to v0.11.1 coming up we will need to update it to include ENRs as being the primary basis for node identities.  Due to discovery now being primarily based on discoveryV5 instead of libp2p's kademlia dht, this change will be required. 

Things needed to be added to the `peerStatus` struct would be the peer's ENR and peer's metadata . Using the metadata sequence number we can validate whether attestation subnets are old/current. This issue is to track the resolution of the above.

### PR DESCRIPTION
**Pull request by nisdas**
- [x] Add new methods to peer status handler
- [x] Add new metadata methods for individual peers.
- [x] Use ENRs as basis to store initial records of peers
- [x] Batch together our status request and ping request into a single method.
- [x] Handle differing metadata sequence numbers.
- [x] Add tests

Resolves #5290

### CODE REVIEW

File: beacon-chain/p2p/testing/mock_peersprovider.go [hunk 1]
```diff
@@ -3,6 +3,8 @@ package testing
 import (
 	"sync"
 
+	"github.com/ethereum/go-ethereum/p2p/enr"
+
```
Comment (nisdas) [inline beacon-chain/p2p/testing/mock_peersprovider.go:7]:
```suggestion
```

File: beacon-chain/p2p/peers/status_test.go [hunk 2]
```diff
@@ -4,8 +4,13 @@ import (
 	"bytes"
 	"crypto/rand"
 	"fmt"
+	"reflect"
 	"testing"
 
+	"github.com/prysmaticlabs/go-bitfield"
+
```
Comment (nisdas) [inline beacon-chain/p2p/peers/status_test.go:11]:
```suggestion
```

File: beacon-chain/p2p/peers/status_test.go [hunk 3]
```diff
@@ -4,8 +4,13 @@ import (
...
+	"github.com/ethereum/go-ethereum/p2p/enr"
+
```
Comment (nisdas) [inline beacon-chain/p2p/peers/status_test.go:13]:
```suggestion
```

File: beacon-chain/p2p/testing/p2p.go [hunk 4]
```diff
@@ -8,6 +8,8 @@ import (
 	"testing"
 	"time"
 
+	"github.com/ethereum/go-ethereum/p2p/enr"
+
```
Comment (nisdas) [inline beacon-chain/p2p/testing/p2p.go:12]:
```suggestion
```

File: beacon-chain/sync/initial-sync-old/round_robin_test.go [hunk 5]
```diff
@@ -8,6 +8,8 @@ import (
 	"testing"
 	"time"
 
+	"github.com/ethereum/go-ethereum/p2p/enr"
+
```
Comment (nisdas) [inline beacon-chain/sync/initial-sync-old/round_robin_test.go:12]:
```suggestion
```

File: beacon-chain/sync/initial-sync/round_robin_test.go [hunk 6]
```diff
@@ -8,6 +8,8 @@ import (
 	"testing"
 	"time"
 
+	"github.com/ethereum/go-ethereum/p2p/enr"
+
```
Comment (nisdas) [inline beacon-chain/sync/initial-sync/round_robin_test.go:12]:
```suggestion
```

File: beacon-chain/sync/pending_attestations_queue_test.go [hunk 7]
```diff
@@ -6,6 +6,8 @@ import (
 	"testing"
 	"time"
 
+	"github.com/ethereum/go-ethereum/p2p/enr"
+
```
Comment (nisdas) [inline beacon-chain/sync/pending_attestations_queue_test.go:10]:
```suggestion
```

File: beacon-chain/sync/pending_blocks_queue_test.go [hunk 8]
```diff
@@ -5,6 +5,8 @@ import (
 	"sync"
 	"testing"
 
+	"github.com/ethereum/go-ethereum/p2p/enr"
+
```
Comment (nisdas) [inline beacon-chain/sync/pending_blocks_queue_test.go:9]:
```suggestion
```

File: beacon-chain/sync/rpc_ping.go [hunk 9]
```diff
@@ -22,13 +22,24 @@ func (r *Service) pingHandler(ctx context.Context, msg interface{}, stream libp2
...
-	if err := r.validateSequenceNum(*m, stream.Conn().RemotePeer()); err != nil {
+	changed, err := r.validateSequenceNum(*m, stream.Conn().RemotePeer())
+	if err != nil {
 		return err
 	}
+	if changed {
+		// send metadata request in a new routine and stream.
+		go func() {
+			md, err := r.sendMetaDataRequest(ctx, stream.Conn().RemotePeer())
+			if err == nil {
```
Comment (rauljordan) [inline beacon-chain/sync/rpc_ping.go:33]: Is there any other handling of this error we should do? maybe logging it?

File: beacon-chain/p2p/service.go [hunk 10]
```diff
@@ -533,6 +536,15 @@ func (s *Service) addKadDHTNodesToExclusionList(addr string) error {
 	return nil
 }
 
+func (s *Service) updateBitfield(bitV bitfield.Bitvector64) {
```
Comment (terencechain) [inline beacon-chain/p2p/service.go:539]: should this be next to `retrieveIndicesFromBitfield` where `status.go` is?  or `subnets.go`

File: beacon-chain/sync/rpc_ping.go [hunk 11]
```diff
@@ -93,6 +93,9 @@ func (r *Service) validateSequenceNum(seq uint64, id peer.ID) (bool, error) {
 	if err != nil {
 		return false, err
 	}
+	if md == nil {
+		return false, nil
```
Comment (nisdas) [inline beacon-chain/sync/rpc_ping.go:97]: this should actually return true then, we will need to request the peer's metadata in that case

**Pull request merged**